cmd/zpool/zpool_main.c

```diff
@@ -2306,8 +2305,8 @@ print_status_trim(vdev_stat_t *vs, boolean_t verbose)
 				    100 / (vs->vs_trim_bytes_est + 1));
 			}
 
-			(void) localtime_r(&t, &zaction_ts);
-			(void) strftime(tbuf, sizeof (tbuf), "%c", &zaction_ts);
+			(void) ctime_r(&t, tbuf);
+			tbuf[24] = 0;
```

Comment:
From what I can gather `strftime` is recommended these days instead of `ctime` specifically because it's locale-sensitive. 

> ctime returns a pointer to static data and is not thread-safe. In addition, it modifies the static [tm](https://en.cppreference.com/w/c/chrono/tm) object which may be shared with [gmtime](https://en.cppreference.com/w/c/chrono/gmtime) and [localtime](https://en.cppreference.com/w/c/chrono/localtime). POSIX marks this function obsolete and recommends [strftime](https://en.cppreference.com/w/c/chrono/strftime) instead. The C standard also recommends [strftime](https://en.cppreference.com/w/c/chrono/strftime) instead of ctime and ctime_s because strftime is more flexible and locale-sensitive.  From https://en.cppreference.com/w/c/chrono/ctime

We even use `gettext()` here for the non-data portion but don't provide any translations for the english text.  Removing the locale awareness here seems like the wrong way to go, but I agree the current behavior is also quite undesirable.

I see there are also several other places which make the same assumptions.  If we're going to consider changing this, we'll want to at least be consistent.  Suggestions welcome.

Reply:
Originally I suggested Maxim to stay with strftime(), but use format that exactly matches LANG=C format for date instead of %c.

Reply:
I couldn't find the specific format string for `LANG=C` in `man strftime`

Reply:
It is defined in the C documentation. C11 document  says:

`%c equivalent to ‘‘%a %b %e %T %Y’’.`

Reply:
Ah, but those will be substituted to nationalized names according to running locale. Seems like using ctime() is the most correct approach until ZFS tools are fully localized.

Reply:
It does seem like the least bad way to handle this for now is to use `ctime_r` in all of the places where `%c` is used with `strftime()`.  A quick grep of the source turned up four locations.  @part1zano can you update this PR to switch all of them.

```
cmd/zpool/zpool_main.c:		(void) strftime(tbuf, sizeof (tbuf), "%c", &zaction_ts);
cmd/zpool/zpool_main.c:		(void) strftime(tbuf, sizeof (tbuf), "%c", &zaction_ts);
lib/libzfs/libzfs_pool.c:	    strftime(timestr, 128, "%c", &t) != 0) {
lib/libzfs/libzfs_pool.c:	    strftime(timestr, 128, "%c", &t) != 0) {
```

Reply:
@behlendorf 
Yep, I'm on it.

Reply:
@behlendorf please take a look.

Reply:
@behlendorf please review